test/framework/src/main/java/org/elasticsearch/index/RandomCreateIndexGenerator.java

```diff
@@ -81,7 +83,7 @@ public static Settings randomIndexSettings() {
      * Creates a random mapping, with no mention of types.
      */
     public static XContentBuilder randomMapping() throws IOException {
-        XContentBuilder builder = XContentFactory.jsonBuilder();
+        XContentBuilder builder = XContentFactory.contentBuilder(randomFrom(XContentType.values()));
```

Comment:
This might have side-effects on other tests using this. Its good that its tested though. Maybe make sure to run a few of the tests that somehow use this with a couple of repetitions locally (@Repeat) just to be sure and not be suprised by test failures some time later on CI.